### DISCUSSION TITLE
[Run issue]:Elekta phase space

### DISCUSSION THREAD

**Discussion by bradyhu78** [OpenTOPAS/OpenTOPAS#39] [Run-Problem] [answered]
### Contact Details

784745021@qq.com

### What was the issue?

The ELEKTA phase space file downloaded from IAEA has four parts. Is it separate? Or do I need to splice it, if so, could someone tell me how to splice it  Or share some relevant information with me ?when i simulate the MLINAC with phase space file(only part 1), There seems to be something wrong with it, thank you!

### TOPAS category

Geometry

### Version

4.0

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Attestation

- [X] I testify that I have fully read the bug reporting guideline beforehand.

**Comment by JoseRamosMendez** [comment, marked as answer]
It looks like you can read each each part of the phase space independently. Please see this example [examples/PhaseSpace/ReadElekta.txt](https://github.com/OpenTOPAS/OpenTOPAS/blob/master/examples/PhaseSpace/ReadElekta.txt)
Best!

**Reply by bradyhu78** [reply]
yes, I know that, I mean if I want to simulate the Elekta machine in real, should i use the four parts of phase space file so that i can get the reliable result?

**Reply by bradyhu78** [reply]
If I want to use the four parts of phase space file in one simulation, how should i do? thank you!
Best

**Reply by JoseRamosMendez** [reply]
Phase spaces have an intrinsic statistical uncertainty and, based on your simulation setup (voxel resolution), you won't reach a better uncertainty than that of the phase space. The more statistics the better. Therefore, using all parts will ensure a lower uncertainty. But, to get a "real" scenario, the number of histories is prohibitively large. So, in MC applications one run sufficient histories to reach a desired statistical uncertainty, and just scale (e.g., dose) from that. For radiotherapy research 0.5%, 1-2 stdv (well below experimental uncertainties) should be the goal. 

To determine how many histories (hence phase spaces) you might try

- For a given voxel resolution, play a decent amount of histories from the phase space
- Get the statistical uncertainty of your quantity of interest
- Use the law of large numbers to determine the number of histories that you'll require to achieve a 0.5% (ideal for research purposes) uncertainty, that is, the standard deviation drops with 1/sqrt(Nh)
- With that Nh you will figure out if you need to play (and replay) all the phase spaces
- Keep track of the computation time so that you can estimate how much resources you'll need
- You will find out if you need to reuse multiple times each phase space, so have a look at this useful [paper](http://www.ncbi.nlm.nih.gov/pubmed/11324958) 

Good luck!

**Reply by bradyhu78** [reply]
Your in-depth response is incredibly valuable to me. Many thanks

**Reply by JoseRamosMendez** [reply]
You can use as many sources as you want, just copy/paste and rename the sources accordingly. E.g.,

    s:So/MyPhaseSpace1/Type = "PhaseSpace"
    ...
    s:So/MyPhaseSpace1/PhaseSpaceFileName = "PhaseSpaceFileName1"

    s:So/MyPhaseSpace2/Type = "PhaseSpace"
    ...
    s:So/MyPhaseSpace2/PhaseSpaceFileName = "PhaseSpaceFileName2"

Cheers

**Reply by bradyhu78** [reply]
I got it! Thanks!
